src/generator/generator.ts

```diff
@@ -13,30 +16,26 @@ class Generator {
         const parameters = new Array(tree.params.length).fill(i32);
```

Comment:
Move this down to line 24 and use parameterMapping.size to have a consistent solution.